dbms/src/Columns/ColumnDecimal.cpp

```diff
@@ -230,35 +248,49 @@ void ColumnDecimal<T>::serializeToPosForColumnArrayImpl(
             if (pos[i] == nullptr)
                 continue;
         }
+
         size_t len = array_offsets[start + i] - array_offsets[start + i - 1];
-        if (len <= 4)
+        if constexpr (ensure_unique && is_Decimal256)
         {
             for (size_t j = 0; j < len; ++j)
-                tiflash_compiler_builtin_memcpy(
-                    pos[i] + j * sizeof(T),
-                    &data[array_offsets[start + i - 1] + j],
-                    sizeof(T));
+                pos[i] = serializeDecimal256Helper(pos[i], data[array_offsets[start + i - 1] + j]);
         }
         else
         {
-            inline_memcpy(pos[i], &data[array_offsets[start + i - 1]], len * sizeof(T));
+            if (len <= 4)
```

Comment:
Please add some comments here to explain why is 4 here

Reply:
It's not my code. @gengliqi can you help to explain?

Reply:
It's just a simple optimization. If the length is very small, copying them one by one is faster than std::memcpy.